### ISSUE TITLE
False positives with `>`

### ISSUE THREAD

**Issue by bensaufley** [csstools/stylelint-prefer-nesting#22]
# Problem

I've got PostCSS that looks like this with the `nesting` plugin, making use of the [`>`](https://developer.mozilla.org/en-US/docs/Web/CSS/Child_combinator) child combinator:

```css
.foo {
  > button {
    color: green;
  }

  &.warn > button {
    color: red;

    &:hover,
    &:focus {
      color: blue;
    }
  }
}

.bar {
  button {
    color: yellow;
  }

  > button {
    color: blue;
  }
}
```

I don't see a deeper way to nest these, but this plugin wants me to.

# Expected

No violation, no change. Output of CSS when nesting is resolved should look like:

```css
.foo > button {
  color: green;
}

.foo.warn > button {
  color: red
}

.foo.warn > button:hover,
.foo.warn > button:focus {
  color: blue;
}

.bar button {
  color: yellow;
}

.bar > button {
  color: blue;
}
```

# Encountered

As mentioned above, I see a rule violation on `&.warn > button` and the latter `> button`. When I use auto-fix, this is what I get:

```css
.foo {
  > button {
    color: green;

    &.warn & {
      color: red;

      &:hover,
      &:focus {
        color: blue;
      }
    }
  }
}

.bar {
  button {
    color: yellow;

    > & {
      color: blue;
    }
  }
}
```

This is obviously not an equivalent structure. This will boil down to:

```css
.foo > button {
  color: green
}
.foo > button.warn .foo > button {
  color: red
}
.foo > button.warn .foo > button:hover,
.foo > button.warn .foo > button:focus {
  color: blue;
}

.bar button {
  color: yellow
}
> .bar button { /* I don't think this is even valid CSS? */
  color: blue;
}
```

# Proposed solution

I can't say exactly what's going on under the hood here but there seems to be a fundamental misunderstanding of the `>` combinator. Obviously I would expect _this_ to be a violation and correctly fixed:

```css
.foo { color: green; }
.foo > .bar { color: red; }

/* becomes: */
.foo {
  color: green;

  > .bar {
    color: red;
  }
}
```

But it seems like something extra is happening and I'm not exactly sure what needs to change there so that correct nesting can still happen while this false positive does not.

**Comment by romainmenke** [comment]
This has been fixed in 6.0.2:

https://stylelint.io/demo/#N4Igxg9gJgpiBcIB0AzCEAEwA6A7DGAfBgEYCuALhRPjvgRpADYQBO8GA5qzDLgNx4CAXzxCMAMiQB3AIat8xclRpZxBZmw48og3OsnwAFhABuMVgBoDE+GjBkAzmvoNGEFu1JMyMPW9F6QMC8JBJ5FwJlaloDTS8ATxgmFml-DBD6JUoYyIZ4jhIfP3FgkEsQFABLJhgAOVkAWzhEGAAPJoAHWqQwR0dy8BpqzgQQds62CgxYFFkyJmm6AmwQbrJOKtxHVY4AbQNVxwoE2qYtigBaJxhL3Bhjrc5V8QBda3pV1gWH3byV8D9ageRwAehudweFCef1WsiYcgSOxAGAAPqiMKsqpxcGwYC8gnhhHpBpBcCMAGJsRqyChjY6nZIXXrDbFIRoAKwGFVgnQGiGWmJADLOF1hICYtKhqw+AJFTNwVzJI0ux1kuCg8ig4slFGl5XERxOosV10ct3uj1wzwQQoAbEgAAxIABMBMCIGEQA
